browser/blueprints.py

```diff
@@ -33,6 +33,29 @@ def health():
     return jsonify({'message': 'success'}), 200
 
 
+class InvalidExtension(Exception):
+    status_code = 400
+
+    def __init__(self, message, status_code=None, payload=None):
+        Exception.__init__(self)
+        self.message = message
+        if status_code is not None:
+            self.status_code = status_code
+        self.payload = payload
+
+    def to_dict(self):
+        rv = dict(self.payload or ())
+        rv['message'] = self.message
+        return rv
+
+
+@bp.errorhandler(InvalidExtension)
+def handle_invalid_usage(error):
+    response = jsonify(error.to_dict())
+    response.status_code = error.status_code
+    return response
+
+
 @bp.errorhandler(Exception)
 def handle_exception(error):
     """Handle all uncaught exceptions"""
```

Comment:
How does Flask cascade these errors? Is it determined by the declaration order?

Reply:
From [https://flask.palletsprojects.com/en/1.1.x/errorhandling/#handling](https://flask.palletsprojects.com/en/1.1.x/errorhandling/#handling):

>When an exception is caught by Flask while handling a request, it is first looked up by code. If no handler is registered for the code, it is looked up by its class hierarchy; the most specific handler is chosen.